macros/src/actix.rs

```diff
@@ -719,10 +720,13 @@ pub fn emit_v2_definition(input: TokenStream) -> TokenStream {
     };
 
     let schema_name = name.to_string();
+    let random_prefix = generate_random_uppercase(6);
```

Comment:
not sure this is the right thing to do here, otherwise we're generating different specs on different builds?

Reply:
That is a good point. That is indeed the case here. For what I've read and heard, it's not possible to get the fully qualified path in a proc macro.

Do you have any suggestions for another solution?

Reply:
Even a module name is probably not ideal as we want the type name to be named somewhat appropriately, since it's part of an API - what about a way to explicitly rename the openApi type? eg:
```rust
#[derive(Deserialize, Serialize, Apiv2Schema)]
#[apiv2(rename = "PetThatSings")]
struct Pet { }
```
It would also be helpful if we could issue a warning or error if we encounter a duplicate when stitching up the spec.

Reply:
That could also definitely be a good option, though I don't know how to implement this myself. Could we maybe 'compute' a hash of sorts and use that as prefix? That way the name is consistent and only changes when the struct itself does.

This means a user isn't required to rename every struct, which gets tedious, but it doesnt leave everything random either.

Reply:
I have figured out a way to get the fully qualified path, so that is now too an option

Reply:
How would the type names look like? 
That would be better than random, but that's still something we wouldn't want to do by default. Maybe we could have an attribute enables using that fqp?
```rust
#[derive(Deserialize, Serialize, Apiv2Schema)]
#[apiv2(prefix_fqp)]
struct Pet { }
or 
#[derive(Deserialize, Serialize, FQPApiv2Schema)]
struct Pet { }
```

Reply:
I haven't generated it out yet, but by default we get e.g foo::bar::Request, but we could emit those double colons (it probably does not work well with yaml, so we should I think), or replace them with e.g hyphens.

An attribute that enables it would be the best solution I think, additionally we could stick it behind a feature flag too.